File: problem_007.pl
#!/usr/bin/env perl

print "Project Euler - Problem 7:\n";
print "What is the 10,001st prime number?\n\n";

sub isPrime{
    local $i, $n;
    my($n) = @_;

    if ($n < 2){
        return 0;
    }

    if ($n == 2){
        return 1;
    }

    if ($n % 2 == 0){
        return 0;
    }

    for ($i=3 ; $i ** 2 <= $n ; $i+=2){
        if ($n % $i == 0){
            return 0;
        }
    }

    return 1;
}

$n = 3;
$c = 2;

while ($c < 10001){
    $n += 2;
    if (isPrime($n) == 1){
        $c++;
    }
}

print $n, "\n";

File: problem_007.js
#!/usr/bin/node

console.log("Project Euler - Problem 7:");
console.log("What is the 10,001st prime number?\n");

const target = 10001;
var primes = [2];

function isPrime(n, primes) {
    for (var p of primes)
	if (n % p == 0)
	    return false;

    return true;
}

var n = 3;
while (primes.length < target) {
    if (isPrime(n, primes))
	primes.push(n);
    n += 2;
}

console.log(primes[primes.length - 1]);
